src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs

```diff
@@ -436,7 +436,9 @@ public static partial class ImmutableDictionary
         public static System.Collections.Immutable.ImmutableDictionary<TKey, TValue>.Builder CreateBuilder<TKey, TValue>(System.Collections.Generic.IEqualityComparer<TKey>? keyComparer) where TKey : notnull { throw null; }
         public static System.Collections.Immutable.ImmutableDictionary<TKey, TValue>.Builder CreateBuilder<TKey, TValue>(System.Collections.Generic.IEqualityComparer<TKey>? keyComparer, System.Collections.Generic.IEqualityComparer<TValue>? valueComparer) where TKey : notnull { throw null; }
         public static System.Collections.Immutable.ImmutableDictionary<TKey, TValue> CreateRange<TKey, TValue>(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<TKey, TValue>> items) where TKey : notnull { throw null; }
+        public static System.Collections.Immutable.ImmutableDictionary<TKey, TValue> CreateRangeWithOverwrite<TKey, TValue>(params System.ReadOnlySpan<System.Collections.Generic.KeyValuePair<TKey, TValue>> items) where TKey : notnull { throw null; }
```

Comment:
I missed API review so I might be lacking context -- I take it that these APIs were added because because dictionary expressions are meant to have overwrite semantics? I should caution that this could [spill over into serialization](https://github.com/dotnet/runtime/issues/107787) where overwrite semantics is implicated in [known security vulnerabilities](https://github.com/dotnet/runtime/issues/108521).

While it is true that collection literals != collection deserialization and we could choose to special case ImmutableDictionary deserialization, this wouldn't apply to custom collection types with CBA annotations which would also have overwrite semantics.

Reply:
I don't see how this relates to serialization/deserialization.

Anyone already could have done:
```C#
ImmutableDictionary<...> d = ImmutableDictionary<...>.Empty.SetItems(source);
```
and serialized that.

Deserialization also isn't impacted by this method in any way. Deserialization can still enjoy throw-on-duplicates as it currently is.

The overwrite behavior is simply the behavior chosen by dictionary expressions, so if you construct a dictionary using that, you're getting set rather than add behavior on each element.

Reply:
> I don't see how this relates to serialization/deserialization

Specifically this part in the [linked issue](https://github.com/dotnet/runtime/issues/107787):

> This issue proposes that we extend generic support for collection type deserialization by detecting the following patterns (in order of preference):
>
> 2. Collection declares a span factory via CollectionBuilderAttribute or,

Assuming we adopt the convention that user-defined dictionary types with CBA annotation are deserialized using the specified factory and we have specified that dictionary expressions must have overwrite semantics, then it follows that deserializing such collections would also have overwrite semantics.

Reply:
FrozenDictionary already uses overwrite behavior when constructing one.

https://learn.microsoft.com/en-us/dotnet/api/system.collections.frozen.frozendictionary.tofrozendictionary?view=net-9.0#system-collections-frozen-frozendictionary-tofrozendictionary-2(system-collections-generic-ienumerable((system-collections-generic-keyvaluepair((-0-1))))-system-collections-generic-iequalitycomparer((-0)))

If we don't think that behavior is appropriate for JsonSerializer, then JsonSerializer shouldn't use CollectionBuilder annotations, which can be implemented to do anything.